### PR TITLE
Add framework in primary tagging in edit entries

### PR DESCRIPTION
**Pull request by AdityaKhatri**
## Changes

* Fetch and display primary tagging in entry edit page

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: app/views/Project/EntryEdit/index.tsx [hunk 1]
```diff
@@ -149,30 +203,49 @@ function EntryEdit(props: Props) {
                                 ready={ready}
                                 pending={leadGetPending}
                                 leadInitialValue={leadInitialValue}
-                                projectId={projectId}
+                                projectId={+projectId}
```
Comment (tnagorra) [inline app/views/Project/EntryEdit/index.tsx:200]:
```suggestion
                                projectId={projectId ? +projectId : undefined}
```

File: app/views/Project/EntryEdit/index.tsx [hunk 2]
```diff
@@ -38,11 +57,45 @@ function EntryEdit(props: Props) {
     const { className } = props;
     const { project } = React.useContext(ProjectContext);
     const { leadId } = useParams<{ leadId: string }>();
-    const projectId = project ? +project.id : undefined;
+    const projectId = project ? project.id : undefined;
+    const frameworkId = project?.analysisFramework?.id;
```
Comment (tnagorra) [inline app/views/Project/EntryEdit/index.tsx]: We should update the permission such that if there is not analytical framework in the project the page should not be accessible.

File: app/views/Project/EntryEdit/index.tsx [hunk 3]
```diff
@@ -22,14 +27,28 @@ import {
...
 import Review from './Review';
 
 import styles from './styles.css';
 
+type FrameworkDetailsMini = Pick<AnalysisFramework, 'id' | 'primaryTagging' | 'secondaryTagging'>;
+/*
+type Entry = NonNullable<
+NonNullable<NonNullable<NonNullable<ProjectFrameworkQuery['project']>['lead']>['entries']>[number]
+>;
+*/
```
Comment (tnagorra) [inline app/views/Project/EntryEdit/index.tsx:50]: Let's remove this if not required

File: app/views/Project/EntryEdit/queries.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            key
+            order
+            title
+            properties
+            widgetId
+            width
+        }
+        id
+    }
+`;
```
Comment (tnagorra) [inline app/views/Project/EntryEdit/queries.ts]: Why create a fragment?

Let's also try to not create a separate file for query.
    - Cleanup gets harder
    - Others will add any query inside this file.

File: app/views/Project/EntryEdit/PrimaryTagging/Sections/styles.css [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            }
+
+            .tab {
+                flex-grow: 1;
+
+                /*
+                &.errored {
+                    color: var(--dui-color-danger);
+                }
+                */
```
Comment (tnagorra) [inline app/views/Project/EntryEdit/PrimaryTagging/Sections/styles.css:27]: Why comment this out?

File: app/components/framework/FrameworkOutput/index.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+function Canvas<T>(props: Props<T>) {
+    const {
+        name,
+        /*
+        widgets,
+        onWidgetDelete,
+        onWidgetEdit,
+        onWidgetOrderChange,
+        */
```
Comment (tnagorra) [inline app/components/framework/FrameworkOutput/index.tsx]: Any reason to keep these comments?

File: app/components/framework/FrameworkOutput/index.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        isSecondary = false,
+        error: riskyError,
+    } = props;
+
+    const error = getErrorObject(riskyError);
+
+    const handleWidgetValueChange = useCallback(
+        (_: unknown, widgetName: string) => {
+            // NOTE: when we start work no tagging page, we need to handle this
+            // for preview page, we can skip this as the components are disabled any way
```
Comment (tnagorra) [inline app/components/framework/FrameworkOutput/index.tsx]:
```suggestion
            // NOTE: when we start work on tagging page, we need to handle this
            // for preview page, we can skip this as the components are disabled any way
```

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: app/views/Project/EntryEdit/index.tsx (hunk 1 above)
Comment (AdityaKhatri) [reply app/views/Project/EntryEdit/index.tsx:200]: This is not necessary

File: app/views/Project/EntryEdit/index.tsx (hunk 3 above)
Comment (AdityaKhatri) [reply app/views/Project/EntryEdit/index.tsx:50]: Will be required immediately in next PR

File: app/components/framework/FrameworkOutput/index.tsx [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+import { Widget } from '#types/newAnalyticalFramework';
+import WidgetPreview from '#components/framework/WidgetPreview';
+
+import styles from './styles.css';
+
+interface WidgetProps {
+    className?: string;
+    widget: Widget;
+    clientId: string;
+    onWidgetValueChange: (value: unknown, widgetName: string) => void;
```
Comment (tnagorra) [inline app/components/framework/FrameworkOutput/index.tsx]: Why is this type unknown? This should directly be extended from WidgetPreview instead.

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Pull request merged**